apis/v1alpha1/opentelemetrycollector_webhook.go

```diff
@@ -109,5 +109,16 @@ func (r *OpenTelemetryCollector) validateCRDSpec() error {
 		}
 	}
 
+	// validate autoscale with horizontal pod autoscaler
+	if r.Spec.MaxReplicas != nil {
+		if r.Spec.MaxReplicas == nil || *r.Spec.MaxReplicas < int32(1) {
+			return fmt.Errorf("the OpenTelemetry Spec autoscale configuration is incorrect, maxReplicas should be defined and more than one")
+		}
+
+		if r.Spec.Replicas != nil && *r.Spec.Replicas > *r.Spec.MaxReplicas {
```

Comment:
while we are at this, could you please set the `Replicas` in the defaulting webhook if it is null and remove other code that sets the replicas outside of the defaulting webhook?

Reply:
not sure that I got this right, do you ask to set Spec.Replicas equal to 1 in Default() if r.Spec.Replicas == nil? Because I can see where we set Spec.Replicas outside of it, only in tests and here:
```go
	// if autoscale is enabled, use replicas from current Status
		if params.Instance.Spec.MaxReplicas != nil {
			currentReplicas := existing.Status.Replicas
			// if replicas (minReplicas from HPA perspective) is bigger than
			// current status use it.
			if *params.Instance.Spec.Replicas > currentReplicas {
				currentReplicas = *params.Instance.Spec.Replicas
			}
			updated.Spec.Replicas = &currentReplicas
		}
```

Reply:
The replicas should be set to 1 in deployment if it is nil. 

It's fine I will take a look at it in another PR.